index.js

```diff
@@ -537,21 +546,52 @@ class Grid extends react.Component {
         }, SETTINGS_ICON),
         // End of marketplace-header__right
         ),
-            // TODO: Add search bar and sort functionality
-            // react.createElement("div", {
-            //     className: "searchbar--bar__wrapper",
-            // }, react.createElement("input", {
-            //     className: "searchbar-bar",
-            //     type: "text",
-            //     placeholder: "Search for Extensions?",
-            // })),
+        react.createElement("div", {
+            className: "searchbar--bar__wrapper",
+        }, react.createElement("input", {
+            className: "searchbar-bar",
+            type: "text",
+            placeholder: "Search",
+            value: this.state.searchValue,
+            onChange: (event) => {
+                this.setState({ searchValue: event.target.value });
+                searchQuery = event.target.value;
+            },
+            onKeyDown: (event) => {
+                if (event.key === "Enter") {
+                    this.setState({ endOfList: false });
+                    endOfList = false;
+                    this.newRequest(ITEMS_PER_REQUEST);
+                    requested = true;
+                } else if ( // Refreshes result when user deletes all queries
+                    ((event.key === "Backspace") || (event.key === "Delete")) &&
+                    requested &&
+                    this.state.searchValue.trim() === ""
+                ) {
+                    this.setState({ endOfList: false });
+                    endOfList = false;
```

Comment:
I'm noticing a local variable and state variable for `endOfList` in a bunch of places. Is there a reason why?

Reply:
I used it to trigger the loading icon in my latest commit. Otherwise, I'm pretty sure it was implemented before I made this PR? So I'm not entirely sure.

Reply:
Worked without reassigning local variable.

Reply:
Oh, right, yeah, it's one of those globals that were just sitting around. Glad it works with the state. Makes me wonder why it has a local at all though lol. We can figure out later.